web/src/main/webapp/table.ts

```diff
@@ -41,6 +44,14 @@ import {Dialog} from "./dialog";
 // return an enum from a string.
 export type ContentsKind = "Category" | "Json" | "String" | "Integer" | "Double" | "Date" | "Interval";
 
+export function isNumeric(kind: ContentsKind): boolean {
+    return kind == "Integer" || kind == "Double";
+}
+
+export function isCategorical(kind: ContentsKind): boolean {
+    return kind == "Category" || kind == "String"
```

Comment:
This is wrong, strings are not categories.
Maybe you can call this isString or something else. But I suspect that you don't really want this function.